ejb3/src/main/java/org/jboss/as/ejb3/component/stateful/StatefulSessionComponent.java

```diff
@@ -235,6 +235,10 @@ public AccessTimeoutDetails getAccessTimeout(Method method) {
     }
 
     public SessionID createSession() {
+        // check to see that StatefulSessionComponent.stop() has not been called concurrent with the request; this can happen on undeploy (see WFLY-5178)
+        if (getCache() == null)
+            throw EjbLogger.ROOT_LOGGER.componentIsShuttingDown();
+
         return this.cache.create().getId();
```

Comment:
this.cache can still be null at this point.  Worse, the cache might be in the process of stopping, which would cause create() to fail in an ugly way.